src/custom/components/AccountDetails/Transaction/index.tsx

```diff
@@ -98,78 +106,99 @@ function ActivitySummary(params: {
-      activity as Order
+    const order = activity as Order
+    const { inputToken, sellAmount, feeAmount, outputToken, buyAmount, validTo, kind, fulfillmentTime } = order
 
     const sellAmt = CurrencyAmount.fromRawAmount(inputToken, sellAmount.toString())
     const feeAmt = CurrencyAmount.fromRawAmount(inputToken, feeAmount.toString())
     const outputAmount = CurrencyAmount.fromRawAmount(outputToken, buyAmount.toString())
-    const limitPrice = formatSmart(getLimitPrice(activity as Order))
-    // const executedPrice = formatSmart(getExecutedPrice(activity as Order))
-    const executedPrice = null
-
-    orderSummary.from = `${formatSmart(sellAmt.add(feeAmt))} ${sellAmt.currency.symbol}`
-    orderSummary.to = `${formatSmart(outputAmount)} ${outputAmount.currency.symbol}`
-    orderSummary.limitPrice = `${limitPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`
-    orderSummary.executedPrice = executedPrice
-      ? `${executedPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`
-      : undefined
-    orderSummary.validTo = new Date((validTo as number) * 1000).toLocaleString()
-    orderSummary.fulfillmentTime = fulfillmentTime ? new Date(fulfillmentTime).toLocaleString() : undefined
-    orderSummary.kind = kind.toString()
+    const buyTokenDecimals = order?.inputToken?.decimals ?? DEFAULT_PRECISION
+    const sellTokenDecimals = order?.inputToken?.decimals ?? DEFAULT_PRECISION
+
+    const limitPrice = formatSmart(
+      getLimitPrice({
+        buyAmount: order.buyAmount.toString(),
+        sellAmount: order.sellAmount.toString(),
+        buyTokenDecimals,
+        sellTokenDecimals,
+        inverted: false, // TODO: handle invert price
+      })
+    )
+
+    const executedPrice =
+      order.status === OrderStatus.FULFILLED
+        ? formatSmart(
+            getExecutionPrice({
+              executedBuyAmount: '1234567890', // TODO: when we mutate to FULLFILLED (after getting the order from the API), we need to save the actual executed amounts
+              executedSellAmount: '1000000000',
+              buyTokenDecimals,
+              sellTokenDecimals,
+              inverted: false, // TODO: Handle invert price
+            })
+          )
+        : undefined
+
+    orderSummary = {
+      ...DEFAULT_ORDER_SUMMARY,
+      from: `${formatSmart(sellAmt.add(feeAmt))} ${sellAmt.currency.symbol}`,
+      to: `${formatSmart(outputAmount)} ${outputAmount.currency.symbol}`,
+      limitPrice: `${limitPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`,
+      validTo: new Date((validTo as number) * 1000).toLocaleString(),
+      executedPrice: executedPrice
+        ? `${executedPrice} ${outputAmount.currency.symbol} per ${sellAmt.currency.symbol}`
+        : undefined,
+      fulfillmentTime: fulfillmentTime ? new Date(fulfillmentTime).toLocaleString() : undefined,
+      kind: kind.toString(),
+    }
+  } else {
+    orderSummary = { ...DEFAULT_ORDER_SUMMARY }
```

Comment:
why the destructuring here? just set the orderSummary as DEFAULT_ORDER_SUMMARY with the type set (`as type`) as I'm assuming you're doing this because of typescript

Reply:
I just destructured it, so i clone the object, but I guess is fine to reuse it. will change

Reply:
it had nothing to do with TS..